mr/src/main/java/org/elasticsearch/hadoop/rest/SearchRequestBuilder.java

```diff
@@ -209,15 +215,19 @@ private String assemble() {
             pref.append("_shards:");
             pref.append(shard);
         }
-        if (local) {
+        if (!preference.isEmpty() || local) {
```

Comment:
```suggestion
        if (!StringUtils.hasText(preference) || local) {
```

Since a user might place a blank space in here

Reply:
Change made in latest commit.